src/skimage/_shared/utils.py

```diff
@@ -1096,3 +1097,121 @@ def as_binary_ndarray(array, *, variable_name):
                 f"safely cast to boolean array."
             )
     return np.asarray(array, dtype=bool)
+
+
+def _prescale_value_range(image, *, mode, stacklevel=3):
+    """Scale the value range of `image` according to the selected `mode`.
+
+    Parameters
+    ----------
+    image : ndarray
+        The image to scale.
+    mode : {'minmax', 'legacy', False} or tuple[float, float], optional
+        Controls the rescaling behavior for `image`.
+
+        ``'minmax'``
+            Normalize `image` between 0 and 1 regardless of dtype. After
+            normalization its minimum and maximum values will be 0 and 1
+            respectively. This is a shorthand for
+            ``prescale=(image.min(), image.max())``.
+
+        ``'legacy'``
+            Normalize only if `image` has an integer dtype, if `image` is of
+            floating dtype, it is left alone. See :ref:`.img_as_float` for
+            more details.
+
+        ``(lower, higher)``
+            Normalize `image` such that ``lower`` and ``higher`` are scaled
+            with ``(img - lower) / (higher - lower)``
+            to 0 and 1 respectively.
+
+        ``False``
+            Don't prescale the value range of `image` at all.
+
```

Comment:
Did we discuss an int-type scaling option independent of `legacy`?   I mean scaling int images only from dtype min -> 0, dtype max -> 1?

Reply:
Could you expand on that? Do you mean an option that only scales integer images to [0, 1] and ignores floats? Would that differentiate signed and unsigned ints? Do you think it would be more useful than confusing?

Reply:
Are we sure this is a popular need? I think implementing "legacy", "none" and "minmax" should be safely worth it. Not as sure about other modes. Especially if we can always add them later. And don't forget, in the meantime users can do their own scaling and use "node".